### PR TITLE
[Cache] leverage Contracts\Cache

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Implementing #28096

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/cache.xml [hunk 1]
```diff
@@ -149,6 +149,9 @@
         <service id="Psr\Cache\CacheItemPoolInterface" alias="cache.app" />
         <service id="Psr\SimpleCache\CacheInterface" alias="cache.app.simple" />
         <service id="Symfony\Component\Cache\Adapter\AdapterInterface" alias="cache.app" />
-        <service id="Symfony\Component\Cache\CacheInterface" alias="cache.app.taggable" />
+        <service id="Symfony\Contracts\Cache\CacheInterface" alias="cache.app" />
+        <service id="Symfony\Contracts\Cache\ExtendedCacheInterface" alias="cache.app" />
```
Comment (Nyholm) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/cache.xml]: Please remove this line.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/cache.xml [hunk 2]
```diff
@@ -149,6 +149,9 @@
...
+        <service id="Symfony\Contracts\Cache\TagAwareCacheInterface" alias="cache.app.taggable" />
+        <service id="Symfony\Contracts\Cache\ExtendedTagAwareCacheInterface" alias="cache.app.taggable" />
```
Comment (Nyholm) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/cache.xml]: It should be `TagAwareCacheInterface`

**Review by Nyholm — CHANGES_REQUESTED** [review]
This needs to be updated

**Comment by nicolas-grekas** [comment]
thanks fixed

File: src/Symfony/Component/Cache/CacheInterface.php [hunk 3]
```diff
@@ -1,37 +0,0 @@
...
- *
- * On cache misses, a callback is called that should return the missing value.
- * It is given two arguments:
- * - the missing cache key
- * - the corresponding PSR-6 CacheItemInterface object,
- *   allowing time-based expiration control.
- *
- * @author Nicolas Grekas <p@tchwork.com>
- */
-interface CacheInterface
```
Comment (Nyholm) [inline src/Symfony/Component/Cache/CacheInterface.php]: For the record: This interface has never been released, so it is okey to remove it.

**Review by Nyholm — APPROVED** [review]
Thank you. 👍

File: src/Symfony/Contracts/Cache/CacheInterface.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Contracts\Cache;
+
+use Psr\Cache\InvalidArgumentException;
+
+/**
+ * Gets and stores items from a cache.
```
Comment (fabpot) [inline src/Symfony/Contracts/Cache/CacheInterface.php]: `Gets/Stores items from/to a cache`

**Review by fabpot — APPROVED** [review]
With one small comment

File: composer.json [hunk 5]
```diff
@@ -114,7 +114,10 @@
         "psr/cache-implementation": "1.0",
         "psr/container-implementation": "1.0",
         "psr/log-implementation": "1.0",
-        "psr/simple-cache-implementation": "1.0"
+        "psr/simple-cache-implementation": "1.0",
+        "symfony/cache-contracts": "1.0",
+        "symfony/service-contracts": "1.0",
+        "symfony/translation-contracts": "1.0"
```
Comment (nicolas-grekas) [inline composer.json]: I realized we should add this. OK to you?

**Pull request merged**